compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingForall.cpp

```diff
@@ -581,7 +581,8 @@ void TileAndDistributeToWorkgroupsUsingForallOpPass::runOnOperation() {
                        return dominanceInfo.properlyDominates(tilableOp, user);
                      }) &&
         (llvm::count_if(op->getUsers(), [&](Operation *user) {
-           return tiledAndFusedOps.contains(user);
+           return !user->isBeforeInBlock(tilableOp) &&
```

Comment:
I am not sure I follow. 

Actually now that I look at this , the condition seems like a hack. What you want to check is that if a producer has a use that is outside of the tiled and fused ops. I dont understand then <2 condition here.

Reply:
I've updated the logic. PTAL.